bundles/org.openhab.binding.ws980wifi/src/main/java/org/openhab/binding/ws980wifi/internal/ws980wifiHandler.java

```diff
@@ -0,0 +1,126 @@
+        ws980wifi wsObject = new ws980wifi(host, port);
+        log.debug("wsObject for refresh created with {}, {}", wsObject.getHost(), wsObject.getPort());
+
+        if (wsObject.refreshValues()) {
+            updateState("tempInside", new QuantityType<>(wsObject.tempInside, SIUnits.CELSIUS));
+            updateState("tempOutside", new QuantityType<>(wsObject.tempOutside, SIUnits.CELSIUS));
+            updateState("tempDewPoint", new QuantityType<>(wsObject.tempDewPoint, SIUnits.CELSIUS));
+            updateState("tempWindChill", new QuantityType<>(wsObject.tempWindChill, SIUnits.CELSIUS));
+            updateState("heatIndex", new QuantityType<>(wsObject.heatIndex, SIUnits.CELSIUS));
+            updateState("humidityInside", new QuantityType<>(wsObject.humidityInside, Units.PERCENT));
+            updateState("humidityOutside", new QuantityType<>(wsObject.humidityOutside, Units.PERCENT));
+            updateState("pressureAbsolut", new QuantityType<>(wsObject.pressureAbsolut, SIUnits.PASCAL));
+            updateState("pressureRelative", new QuantityType<>(wsObject.pressureRelative, SIUnits.PASCAL));
+            updateState("windDirection",
+                    new QuantityType<>(wsObject.windDirection, org.openhab.core.library.unit.Units.DEGREE_ANGLE));
+            updateState("windSpeed", new QuantityType<>(wsObject.windSpeed, Units.KILOMETRE_PER_HOUR));
+            updateState("windSpeedGust", new QuantityType<>(wsObject.windSpeedGust, Units.KILOMETRE_PER_HOUR));
+            updateState("rainLastHour",
+                    new QuantityType<>(wsObject.rainLastHour, org.openhab.core.library.unit.Units.MILLIMETRE_PER_HOUR));
+            updateState("rainLastDay",
+                    new QuantityType<>(wsObject.rainLastDay, org.openhab.core.library.unit.Units.MILLIMETRE_PER_HOUR));
+            updateState("rainLastWeek",
+                    new QuantityType<>(wsObject.rainLastWeek, org.openhab.core.library.unit.Units.MILLIMETRE_PER_HOUR));
+            updateState("rainLastMonth", new QuantityType<>(wsObject.rainLastMonth,
+                    org.openhab.core.library.unit.Units.MILLIMETRE_PER_HOUR));
+            updateState("rainLastYear",
+                    new QuantityType<>(wsObject.rainLastYear, org.openhab.core.library.unit.Units.MILLIMETRE_PER_HOUR));
+            updateState("rainTotal",
+                    new QuantityType<>(wsObject.rainTotal, org.openhab.core.library.unit.Units.MILLIMETRE_PER_HOUR));
+            updateState("lightLevel", new QuantityType<>(wsObject.lightLevel, org.openhab.core.library.unit.Units.LUX));
+            updateState("uvRaw", new QuantityType<>(wsObject.uvRaw, org.openhab.core.library.unit.Units.IRRADIANCE));
+            updateState("uvIndex",
+                    new QuantityType<>(wsObject.uvIndex, org.openhab.core.library.unit.Units.IRRADIANCE));
+            log.debug("refreshValues successfully done");
+        } else {
+            log.debug("refreshValues stops with Error");
```

Comment:
If refresh fails this should be reflected in the thing state. Preferable with message on what or why it failed. Other if the getting updates values doesn't work the user won't know except that the values don't seem to change. Also if refresh fails preferable set the channel value to undef.

Reply:
done